src/main/java/athena/timetable/Timetable.java

```diff
@@ -81,18 +90,216 @@ private void populateTimetable() {
+        if (string.length() == maxLen) {
+            return string;
+        } else if (string.length() < maxLen) {
+            return String.format("%" + -maxLen + "s", string);
+        }
+        return String.format("%s..", string.substring(0, maxLen - 2));
+    }
+
+    /**
+     * Finds a task in a task list that starts at the given hour.
+     *
+     * @param taskList Task list to search through.
+     * @param hour     Starting time of the task.
+     * @return The task that starts at the given hour. Null if doesn't exist.
+     */
+    private Task findTaskAtHour(TaskList taskList, int hour) {
+        for (Task task : taskList.getTasks()) {
+            int taskHour = task.getTimeInfo().getStartTime().getHour();
+            if (taskHour == hour) {
+                return task;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Generates a row for a day in the timetable containing the task information desired.
+     *
+     * @param day            The TimetableDay object for this day.
+     * @param startHour      The starting hour in 24-hour representation.
+     * @param endHour        The ending hour in 24-hour representation.
+     * @param taskInfoWriter A Function that extracts the information desired from a task.
+     * @return A string containing the task information desired.
+     */
+    private String drawTimetableDayRow(TimetableDay day, int startHour, int endHour,
+                                       Function<Task, String> taskInfoWriter) {
+        String row = "";
+
+        for (int hour = startHour; hour < endHour; hour++) {
+            Task task = findTaskAtHour(day.getTaskList(), hour);
+            if (task == null) {
+                row += "          |";
+                continue;
+            }
+
+            int duration = task.getTimeInfo().getDuration();
+            hour += duration - 1;
+            int boxWidth = duration * 11 - 2;
+            row += String.format(" %s|", shortenOrPadString(taskInfoWriter.apply(task), boxWidth));
+        }
+
+        return row;
+    }
+
+    /**
+     * Generates the first row for a day in the timetable, which contains the name of the day and the task names.
+     *
+     * @param day       The TimetableDay object for this day.
+     * @param startHour The starting hour in 24-hour representation.
+     * @param endHour   The ending hour in 24-hour representation.
+     * @return A string containing name of the day and the task names.
+     */
+    private String drawTimetableDayFirstRow(TimetableDay day, int startHour, int endHour) {
+        String dayShortName = day.getDate().getDayOfWeek().toString().substring(0, 3).toUpperCase();
+        String row = "|  " + dayShortName + "  |";
+        row += drawTimetableDayRow(day, startHour, endHour, new Function<Task, String>() {
+            @Override
+            public String apply(Task task) {
+                return task.getName();
+            }
+        });
```

Comment:
not sure if this is going to be flagged in the checkstyle

Reply:
Nope since the checks passed

Reply:
awesome